apps/photos/src/components/PhotoFrame.tsx

```diff
@@ -82,6 +103,148 @@ const PhotoFrame = ({
+        '#325288',
+        '#85B4E0',
+        '#C1A3A3',
+        '#E1A059',
+        '#426165',
+        '#6B77B2',
+        '#957FEF',
+        '#DD9DE2',
+        '#82AB8B',
+        '#9BBBE8',
+        '#8FBEBE',
+        '#8AC3A1',
+        '#A8B0F2',
+        '#B0C695',
+        '#E99AAD',
+        '#D18484',
+        '#78B5A7',
+    ];
+
+    useEffect(() => {
+        const fetchOwnerEmails = async () => {
+            try {
+                const collection: CollectionItem[] =
+                    await getLocalCollections();
+
+                const emailMapping: { [key: number]: string } = {};
+                collection.forEach((item) => {
+                    const { owner } = item;
+                    if (owner.email) {
+                        emailMapping[owner.id] = owner.email;
+                    }
+                });
+
+                setOwnerIdToEmail(emailMapping);
+            } catch (error) {
+                console.error('Error fetching collection:', error);
+            }
+        };
+        fetchOwnerEmails();
+    }, []);
+
+    useEffect(() => {
+        const FetchSharedAlbum = async () => {
+            try {
+                const ownerUser: User = getData(LS_KEYS.USER);
+
+                const collectionList = getLocalCollections();
+
+                const collections = await collectionList;
+
+                const collectionIds = collections
+                    .map((collection) => {
+                        if (
+                            collection.owner.id !== ownerUser.id ||
+                            collection.publicURLs.length > 0
+                        ) {
+                            return collection.id;
+                        }
+                        return null;
+                    })
+                    .filter((id) => id !== null);
+
+                setSharedAlbums(collectionIds);
+            } catch (error) {
+                console.error('Error fetching data:', error);
+            }
+        };
+        FetchSharedAlbum();
+    }, []);
+
+    useEffect(() => {
+        const filesOfSharedAlbums = async () => {
+            try {
+                const resLocalFiles = getLocalFiles();
+                const files = await resLocalFiles;
+
+                const sharedAlbumsFiles = files
+                    .map((files) => {
+                        if (sharedAlbums.includes(files.collectionID)) {
+                            return files.id;
+                        }
+                        return null;
+                    })
+                    .filter((id) => id !== null);
+
+                setSharedAlbumsFiles(sharedAlbumsFiles);
+            } catch (error) {
+                console.error('Error fetching data:', error);
+            }
+        };
+        filesOfSharedAlbums();
+    }, [sharedAlbums]);
+
+    useEffect(() => {
+        const fileIdToName = async () => {
+            try {
+                const resLocalFiles = getLocalFiles();
```

Comment:
This will result in performance issues when we have large number of files.
We can just pass information about userID to email to the PreviewCard.

Inside the previous card. if file's ownerID is not same as collection ownerID and file's ownerID is present inside the map of userId to email list, we can show the avatar.